### PR TITLE
[java] Add max_task_retries settings in actor creation

### PR DESCRIPTION
**Pull request by kira-lin**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/api/src/main/java/io/ray/api/options/ActorCreationOptions.java [hunk 1]
```diff
@@ -131,6 +135,19 @@ public Builder setMaxRestarts(int maxRestarts) {
       return this;
     }
 
+    /**
+     * This specifies the maximum number of times that the actor task can be resubmitted.
+     * The minimum valid value is 0 (default), which indicates that the actor task can't
+     * be resubmited. A value of -1 indicates that an actor task can be resubmited indefinitely.
+     *
+     * @param maxRestarts max number of actor task restarts
```
Comment (jovany-wang) [inline java/api/src/main/java/io/ray/api/options/ActorCreationOptions.java:143]:
```suggestion
     * @param maxTaskRetries max number of actor task restries.
```

**Review by WangTaoTheTonic — APPROVED** [review]

**Comment by jovany-wang** [comment]
Java lint failed. Please use `mvn spotless:apply` to format the code under `ray/java`.

**Comment by jovany-wang** [comment]
DCO failed: https://github.com/ray-project/ray/pull/28295/checks?check_run_id=8242825365

**Comment by kira-lin** [comment]
messed up something, will open a new pr

**Pull request closed without merging**